specs/interop/token-bridging.md

```diff
@@ -0,0 +1,97 @@
+# Token Bridging
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Interface](#interface)
+- [Implementation](#implementation)
+- [L1 Native Tokens](#l1-native-tokens)
+- [Cross Chain `transferFrom`](#cross-chain-transferfrom)
+- [Factory](#factory)
+- [Upgrading Existing `OptimismMintableERC20Token`s](#upgrading-existing-optimismmintableerc20tokens)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+Unified ERC20 token bridging built on top of the messaging protocol
+can enable tokens to be fungible and movable across the superchain.
+An issue with bridging is that if the security model is not standardized,
+then the bridged assets are not fungible with each other depending on
+which bridge solution was used to facilitate the transfer between
+domains.
+
+## Interface
+
+The `ISuperchainERC20` interface creates a standard way to send tokens
+between domains. In contrast to [xERC20][xerc20] where the user calls
+a bridge that has `mint` and `burn` privileges on the cross chain enabled
+tokens, the user instead calls methods on the token contract directly.
+
+[xerc20]: https://www.xerc20.com/
+
+```solidity
+interface ISuperchainERC20 {
+  function bridgeERC20(uint256 amount, uint256 chainId) external;
+  function bridgeERC20To(address to, uint256 amount, uint256 chainId) external;
+  function finalizeBridgeERC20(address to, uint256 amount) external;
+}
+```
+
+## Implementation
+
+An example implementation that depends on deterministic deployments across chains
+for security is provided. This construction builds on top of the [L2ToL2CrossDomainMessenger][l2-to-l2]
+for both replay protection and domain binding.
+
+[l2-to-l2]: ./predeploys.md#l2tol2crossdomainmessenger
+
+```solidity
+contract OptimismSuperchainERC20 is ERC20, ISuperchainERC20 {
+  constructor(string memory _name, string memory _symbol, uint256 _decimals) ERC20(_name, _symbol, _decimals) {}
+
+  function bridgeERC20(uint256 _amount, uint256 _chainId) external {
+    bridgeERC20To(msg.sender, _amount, _chainId);
+  }
+
+  function bridgeERC20To(address _to, uint256 _amount, uint256 _chainId) public {
+    _burn(msg.sender, amount);
+
+    L2ToL2CrossDomainMessenger.sendMessage({
```

Comment:
shouldn't it be `l2ToL2CrossDomainMessenger`?
```suggestion
    l2ToL2CrossDomainMessenger.sendMessage({
```
otherwise, the variable would collide with the reference contract for L2ToL2CrossDomainMessenger